ratis-server-api/src/main/java/org/apache/ratis/server/leader/LogAppender.java

```diff
@@ -137,17 +137,12 @@ default SnapshotInfo shouldInstallSnapshot() {
   void run() throws InterruptedException, IOException;
 
   /**
-   * Similar to {@link #notify()}, wake up this {@link LogAppender} for an event, which can be:
+   * Get the {@link AwaitForSignal} for events, which can be:
    * (1) new log entries available,
    * (2) log indices changed, or
    * (3) a snapshot installation completed.
    */
-  @SuppressFBWarnings("NN_NAKED_NOTIFY")
-  default void notifyLogAppender() {
-    synchronized (this) {
-      notify();
-    }
-  }
```

Comment:
I think we should keep `notifyLogAppender()` as:

```java
  default void notifyLogAppender() {
    getEventAwaitForSignal().signal();
  }
```

to preserve interface compatibility.

It would also allow keeping existing `notifyLogAppender()` calls and `LogAppender::notifyLogAppender` references.